rasa/engine/graph.py

```diff
@@ -260,6 +260,8 @@ class ExecutionContext:
     model_id: Optional[Text] = None
```

Comment:
Just "discovered" the `node_name` and now I'm wondering: Should we call this `node_id` then? And do we even need this? Isn't `node_name` used like an id? Don't both have to be unique because we do things like `message.add_diagnostic_data(self._execution_context.node_name, out.get(DIAGNOSTIC_DATA))` later ?  🤔

Reply:
The thing is we call it `node_name` everywhere through-out the rest of the engine and graph code - so maybe this is not the best place to change it.

Reply:
oh yes sorry - it's exactly the same. We are just now passing it into the graph components.

Reply:
thanks, just wasn't sure I understood this right 🥳